### ISSUE TITLE
[Bug]: terraform tries to delete iops after ebs type changed from gp2 to gp3

### ISSUE THREAD

**Issue by XiaobinYuan** [hashicorp/terraform-provider-aws#27510]
### Terraform Core Version

1.2.1

### AWS Provider Version

4.27.0

### Affected Resource(s)

I upgraded my OpenSearch ebs type from gp2 to gp3 without specifying iops(it's optional) and applied successfully, but then every time I run terraform again, it tries to remove the iops config comes from aws.

```
~ ebs_options {
         ebs_enabled :		true
  -     iops :				16000
         throughput :		500
         volume_size :		6000
         volume_type :		"gp3"

}
```

And terraform doesn't actually remove the iops default value from aws opensearch.

### Expected Behavior

terraform should not try to remove iops default value from AWS if I don't specify iops in my terraform code.

### Actual Behavior

Terraform shows it's going to remove the iops config.

### Terraform Configuration Files

  ```
ebs_options {
    ebs_enabled = var.ebs_volume_size > 0 ? true : false # true
    volume_size = var.ebs_volume_size
    volume_type = var.volume_type # changed from gp2 to gp3
  }
```

### Steps to Reproduce

Deploy aws opensearch with ebs type gp2.
Upgrade to gp3.
re-run terraform plan/apply.

**Comment by jar-b** [comment]
I was able to reproduce this error, but it appears to have been resolved in the `v4.29.0` provider release (specifically in #26503 by making `iops` optional [**and** computed](https://github.com/hashicorp/terraform-provider-aws/blob/v4.29.0/internal/service/opensearch/domain.go#L398)). 

Reproduction documented below.

### Terraform Configuration

```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.27.0" # version reported in issue
      #version = "4.29.0" # version where fix is applied
    }
  }
}

# Configure the AWS Provider
provider "aws" {}

resource "aws_opensearch_domain" "test" {
  domain_name = "jb-test"

  cluster_config {
    instance_type  = "m5.large.search"
    instance_count = 1
  }

  ebs_options {
    ebs_enabled = true
    volume_size = 10
    volume_type = "gp3"
  }
}
```

### Reproduction Steps

1. `terraform init`, then `terraform apply` to create a domain with a GP3 volume type, omitting `iops` and `throughput` attributes.
2. `terraform plan` and observe proposed in-place update
```
$ terraform -v
Terraform v1.3.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.27.0
```
```
$ terraform plan
aws_opensearch_domain.test: Refreshing state... [id=arn:aws:es:us-west-2:012345678901:domain/jb-test]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_opensearch_domain.test will be updated in-place
  ~ resource "aws_opensearch_domain" "test" {
        id               = "arn:aws:es:us-west-2: 012345678901:domain/jb-test"
        tags             = {}
        # (8 unchanged attributes hidden)

      ~ ebs_options {
          - iops        = 3000 -> null
            # (4 unchanged attributes hidden)
        }

        # (8 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
3. Update AWS provider to `v4.29.0`. `terraform init`, then `terraform plan` and observe no diff
```
$ terraform -v
Terraform v1.3.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.29.0
```

```
$ terraform plan
aws_opensearch_domain.test: Refreshing state... [id=arn:aws:es:us-west-2: 012345678901:domain/jb-test]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

We will leave this open for now to allow time to evaluate the reproduction. If no objections are received we will close this as fixed via #26503.

**Comment by jar-b** [comment]
As there are no reports that this bug is still present, marking this issue closed by #26503 (released in `v4.29.0`).
